File: app/src/main/java/com/example/BackgroundLooper.java
package com.example;

import android.os.HandlerThread;
import android.os.Looper;
import android.os.Process;
import android.support.annotation.Nullable;

public final class BackgroundLooper {
	@Nullable
	private static HandlerThread handlerThread;

	public static Looper get() {
		if (handlerThread == null) {
			handlerThread = new HandlerThread("BackgroundHandlerThread", Process.THREAD_PRIORITY_BACKGROUND);
			handlerThread.start();
		}

		return handlerThread.getLooper();
	}
}

File: libs/toolkit-android/src/main/kotlin/br/com/vitorsalgado/example/utils/BackgroundLooper.kt
package br.com.vitorsalgado.example.utils

import android.os.HandlerThread
import android.os.Looper
import android.os.Process

object BackgroundLooper {
  private var handlerThread: HandlerThread? = null

  fun get(): Looper {
    if (handlerThread == null) {
      handlerThread = HandlerThread("BackgroundHandlerThread", Process.THREAD_PRIORITY_BACKGROUND)
      handlerThread!!.start()
    }

    return handlerThread!!.looper
  }
}
